src/CTA.Rules.Actions/ClassActions.cs

```diff
@@ -425,6 +428,93 @@ ClassDeclarationSyntax AddParametersToMethod(SyntaxGenerator syntaxGenerator, Cl
+
+                    bool asyncCheck = method.Modifiers.Any(mod => mod.Text == Constants.AsyncModifier);
+                    bool voidReturn = method.Modifiers.Any(mod => mod.Text == Constants.VoidModifier);
+                    string returnType = "";
+                    if (!voidReturn)
+                    {
+                        returnType = asyncCheck ? Constants.TaskActionResult : Constants.ActionResult;
+                    }
+                    
+                    var newExpression = expression;
+                    if (expression.Contains(Constants.MonolithService + "." + Constants.CreateRequest) && asyncCheck)
+                    {
+                        newExpression = expression.Insert(expression.IndexOf(Constants.MonolithService), Constants.Await + " ");
+                        newExpression = newExpression.Insert(newExpression.IndexOf(Constants.CreateRequest) + Constants.CreateRequest.Length, Constants.AsyncWord);
+                    }
+
+                    newMethod = newMethod.WithBody(null).WithLeadingTrivia(newMethod.GetLeadingTrivia());
+                    newMethod = newMethod.WithBody(SyntaxFactory.Block(SyntaxFactory.ParseStatement(newExpression))).WithLeadingTrivia(newMethod.GetLeadingTrivia());
+                    if (!string.IsNullOrEmpty(returnType))
+                    {
+                        newMethod = newMethod.WithReturnType(SyntaxFactory.ParseTypeName(returnType)).WithLeadingTrivia(newMethod.GetLeadingTrivia());
+                    }
+                    newMethod = newMethod.NormalizeWhitespace();
+
+                    node = node.ReplaceNode(method, newMethod.WithLeadingTrivia(newMethod.GetLeadingTrivia()));
+
+                    allMembers = node.Members.ToList();
+                    allMethods = allMembers.OfType<MethodDeclarationSyntax>().Where(m => m.Modifiers.Any(mod => mod.Text == Constants.Public) && !m.GetLeadingTrivia().ToFullString().Contains(Constants.MonolithServiceComment));
+                }
+
+                return node;
+            }
+            return ReplaceMethodModifiers;
+        }
+        public Func<SyntaxGenerator, ClassDeclarationSyntax, ClassDeclarationSyntax> GetReplaceWebApiControllerMethodsBodyAction(string expression)
+        {
+            ClassDeclarationSyntax ReplaceMethodModifiers(SyntaxGenerator syntaxGenerator, ClassDeclarationSyntax node)
+            {
+                var allMembers = node.Members.ToList();
+                var allMethods = allMembers.OfType<MethodDeclarationSyntax>().Where(m => m.Modifiers.Any(mod => mod.Text == Constants.Public));
+
+                while (allMethods != null && allMethods.Count() > 0)
+                {
+                    var method = allMethods.FirstOrDefault();
+                    MethodDeclarationActions methodActions = new MethodDeclarationActions();
+                    var addCommentActionFunc = methodActions.GetAddCommentAction(Constants.MonolithServiceComment);
```

Comment:
Same as above

Reply:
Removed as discussed. Using new way to store unique method information.